aip/general/0236.md

```diff
@@ -0,0 +1,377 @@
+consequences which severely impact a customer’s overall infrastructure setup.
+
+Previews are a rollout safety mechanism for Policy resources which gives the
+customer the ability to validate the effect of their proposed changes against
+live production traffic prior to the changes going live.
+
+## Guidance
+
+### System changes
+
+The API logic of this mechanism pushes an “experimental” version of a proposed
+change to the upstream system (data plane) while the existing “effective”
+version remains active. Both versions of the configuration are subsequently
+evaluated for incoming traffic, however, only the effective version’s
+configurations actually have the ability to alter traffic. Meanwhile, the
+output of the experimental version is logged so a customer can validate that
+the proposed changes are as intended. This gives the customer the ability to
+fully understand the effects of the new configuration against live traffic
+before replacing the effective configuration with it.
+
+### Non-goals
+
+#### Non-policy resources
+
+This rollout safety mechanism is for policies only.  
+
+A Policy is a customer-owned resource with a set of rules that govern the
+behavior of how other resources admit or deny a target entity (e.g. network
+traffic for firewall rules, general cloud resources with Org Policy). Generally,
+the outcome of a Policy can be evaluated to a specific set of outcomes. For
+example, a FirewallPolicy evaluates to either ALLOW or DENY for incoming traffic
+into a Virtual Network. This lends well to preview mode as a rollout safety
+mechanism as the proposed changes to a Policy can simply be logged instead of
+enforced.
+
+### Supported use-cases
+
+The following use-cases are considered when making a recommendation for API best
+practices of preview mode:
+- Resource: Each entity to be previewed is a gRPC resource.
+- Collection: A collection of objects live on a REST resource and one or many of
+these can be previewed.
+
+Each API recommended below will have a table explaining how it translates to
+each use-case.
+
+### Nested collection called experiments
+
+A new configuration of a Policy which is to be validated using preview mode is
+stored in a nested collection under that Policy resource.
+
+In this AIP, we will use the example of a hypothetical Policy resource called,
+`Policy`.
+
+`/projects/{project}/locations/{location}/policies/{policy}`
+
+The experimental versions of the resource used for preview mode are stored as a
+nested collection under `Policy`. The resource type will follow the convention
+"<resource_name> + Experiment”.
+
+`/projects/{project}/locations/{location}/policies/{policy}/experiments/{experiment}`
+
+A proto used to represent an experiment **must** contain the following fields:
+
+  1. The required top-level fields for a resource, like `name` and `etag`.
+  2. The `Policy` that is being tested itself.
+  3. The `metadata` field for preview.
```

Comment:
According to the proto example below, the field is `preview_metadata`. Please fix.